dbcon/execplan/calpontsystemcatalog.cpp

```diff
@@ -3607,6 +3615,237 @@ const CalpontSystemCatalog::ROPair CalpontSystemCatalog::tableRID(const TableNam
   throw IDBExcept(ERR_TABLE_NOT_IN_CATALOG, args);
 }
 
+// This function is similar to CalpontSystemCatalog::tableRID, except that
+// instead of returning a ROPair for the table, it returns the OID for the
+// AUX column for the table
+CalpontSystemCatalog::OID CalpontSystemCatalog::tableAUXColumnOID(const TableName& tableName,
+                                                                  int lower_case_table_names)
+{
+  TableName aTableName;
+  aTableName.schema = tableName.schema;
+  aTableName.table = tableName.table;
+  if (lower_case_table_names)
+  {
+    boost::algorithm::to_lower(aTableName.schema);
+    boost::algorithm::to_lower(aTableName.table);
+  }
+
+  if (aTableName.schema.compare(CALPONT_SCHEMA) == 0)
+  {
+    std::ostringstream oss;
+    oss << "tableAUXColumnOID() cannot be called on a ";
+    oss << CALPONT_SCHEMA << " schema table";
+    throw runtime_error(oss.str());
+  }
+
+  DEBUG << "Enter tableAUXColumnOID: " << tableName.schema << "|" << tableName.table << endl;
+
+  // look up the OID in the cache first
+  OID auxColOid;
+
+  checkSysCatVer();
+
+  boost::mutex::scoped_lock lk1(fTableAUXColumnOIDMapLock);
+  TableOIDmap::const_iterator iter = fTableAUXColumnOIDMap.find(aTableName);
+
+  if (iter != fTableAUXColumnOIDMap.end())
+  {
+    auxColOid = iter->second;
+    return auxColOid;
+  }
+
+  lk1.unlock();
+
+  // select auxcolumnoid from systable where schema = tableName.schema and tablename = tableName.table;
+  CalpontSelectExecutionPlan csep;
+  CalpontSelectExecutionPlan::ReturnedColumnList returnedColumnList;
+  CalpontSelectExecutionPlan::FilterTokenList filterTokenList;
+  CalpontSelectExecutionPlan::ColumnMap colMap;
+
+  SimpleColumn* c1 =
+    new SimpleColumn(CALPONT_SCHEMA + "." + SYSTABLE_TABLE + "." + AUXCOLUMNOID_COL, fSessionID);
```

Comment:
Not a big deal but we can have a constexpr string saving 12 string additions in the CALPONT_SCHEMA + "." + SYSTABLE_TABLE + "." expression.

Reply:
Or even replace all those expressions with a static constexpr string.

Reply:
Good point, I have replaced these additions with a const string.